File: wrapper-app-front/src/app/dtos/PredmetPredavacDto.ts
import { AsistentZauzeceDto } from "./AsistentZauzeceDto";

export interface PredmetPredavacDto {
    predmetId: string;
    predmetPlan: number;
    predmetOznaka: string;
    predmetNaziv: string;
    predmetGodina: number;
    profesor: string;
    ostaliProfesori: string[];
    asistentZauzeca: AsistentZauzeceDto[]
    block: boolean;
}

File: wrapper-app/src/main/java/com/wrapper/app/dto/response/PredmetPredavacDto.java
package com.wrapper.app.dto.response;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class PredmetPredavacDto {

    private String predmetId;
    private int predmetPlan;
    private String predmetOznaka;
    private String predmetNaziv;
    private int predmetGodina;
    private PredavacDto profesor;
    private List<PredavacDto> ostaliProfesori;
    private List<AsistentiZauzecaDto> asistentZauzeca;
    private boolean block;

    public PredmetPredavacDto() {
        this.ostaliProfesori = new ArrayList<>();
        this.asistentZauzeca = new ArrayList<>();
    }
}
